src/types/cursor/xcursor.rs

```diff
@@ -100,6 +104,15 @@ impl Drop for XCursorTheme {
 }
 
 impl<'theme> XCursor<'theme> {
+    /// NOTE this lifetime is defined by the user of the function, but it must not outlive the
+    /// `XCursorManager` that hosts the xcursor.
+    pub(crate) unsafe fn new<'unbound>(xcursor: *mut wlr_xcursor) -> XCursor<'unbound> {
```

Comment:
Please rename this to `from_ptr` as it's not actually constructing one.

This same function name is used in other parts of the code base to imply that it's constrained by another type. "new" is sort of the constructor in Rust and you aren't actually constructing a new `XCursor`. Usually this distinction is not important, but I do this to make it clearer to show the lifetime relationship in this super unsafe crate.